accord-core/src/main/java/accord/utils/async/AsyncChainCombiner.java

```diff
@@ -96,6 +96,9 @@ private void callback(int idx, I result, Throwable throwable)
             }
             catch (Throwable t)
             {
+                // TODO (correctness): an issue with callbacks vs AsyncResult is that processing the callback may throw.
+                // In this case it isn't known if the callback was already called, so have to call again; leading to
+                // multiple executions
                 callback.accept(null, t);
```

Comment:
I am not sure we are even supposed to provide a callback with its own exceptions. I don't think that is the contract, if a callback throws it should go to the uncaught exception handler or the exception handling for the code that invoked the callback.